app/services/courses/content_status_service.rb

```diff
@@ -1,23 +1,12 @@
 module Courses
   class ContentStatusService
     def execute(enrichment:, recruitment_cycle:)
-      if enrichment.blank?
-        if recruitment_cycle.next?
-          :rolled_over
-        else
-          :empty
-        end
-      elsif enrichment.published?
-        :published
-      elsif enrichment.withdrawn?
-        :withdrawn
-      elsif enrichment.has_been_published_before?
-        :published_with_unpublished_changes
-      elsif enrichment.rolled_over?
-        :rolled_over
-      else
-        :draft
-      end
+      return :rolled_over if (recruitment_cycle.next? && enrichment.blank?) || enrichment&.rolled_over?
+      return :published if enrichment&.published?
+      return :withdrawn if enrichment&.withdrawn?
+      return :published_with_unpublished_changes if enrichment&.has_been_published_before?
```

Comment:
Is there an actually need for this any more?

It looks more complexed then it should be, since it gets remapped later.

Reply:
Yeah it's still used by the enrichment stuff so i think we still need it. Basically the real change is just avoiding sending `empty` and `published_with_unpublished_changes` down the api because the latter is still used internally (which is why it's the only one remapped in the api serializer otherwise whatever value was determined by this service gets returned)